CMakeLists.txt

```diff
@@ -27,7 +35,18 @@ include(cmake/CPM.cmake)
 CPMAddPackage(
   NAME PackageProject.cmake
   GITHUB_REPOSITORY TheLartians/PackageProject.cmake
-  VERSION 1.4
+  VERSION 1.4.1
+)
+
+# Note: If fmt is not imported, this is needed to prevent: CMake Error: install(EXPORT
+# "GreeterTargets" ...) includes target "Greeter" which requires target "fmt" that is not in any
+# export set.  see too https://gitlab.kitware.com/cmake/cmake/-/issues/15415
+set(FMT_VERSION 7.1.3)
+CPMAddPackage(
+  NAME fmt
+  GIT_TAG ${FMT_VERSION}
+  GITHUB_REPOSITORY fmtlib/fmt
+  # XXX OPTION "FMT_INSTALL YES"
 )
```

Comment:
I think the install flag should be enabled, so we can get an installable target? Otherwise the project won't be installable if fmt ever includes compiled sources or is used in a public header. Also I'm not a fan of storing the version in an external variable. I see why we're doing so, as we want to link to the correct version after installation, but there should be a better way. A better option might be to modify CPM.cmake to define `${FMT_VERSION}` for us.

For simplicity let's maybe use the explicit version `7.1.3` twice for now (the CI should already warn if they're not the same) and change it to `${FMT_VERSION}` once CPM.cmake has this functionality.

Reply:
OK, good idea!